openml_OS/models/api/v1/Api_evaluation.php

```diff
@@ -127,53 +133,69 @@ private function evaluation_list($segs, $user_id) {
       $where_limit =  ' LIMIT ' . $offset . ',' . $limit;
     }
     $where_task_closed = ' AND (`t`.`embargo_end_date` is NULL OR `t`.`embargo_end_date` < NOW() OR `r`.`uploader` = '.$user_id.')';
-
+    
     $where_runs = $where_task . $where_setup . $where_uploader . $where_impl . $where_run . $where_tag . $where_task_closed;
+    $where_total = $where_runs . $where_function;
 
     //pre-test, should be quick??
-    if($limit == false) { // skip pre-test if less than 10000 are requested by definition
+    if($limit == false || $per_fold) { // skip pre-test if less than 10000 are requested by definition
       $count = 0;
       //shortcuts
       if (!$implementation_id) {
         $sql_test =
           'SELECT count(distinct r.rid) as count ' .
           'FROM run r, task t '.
           'WHERE r.task_id = t.task_id ' . 
-          $where_runs . $where_limit;
+          $where_runs;
         $count = $this->Evaluation->query($sql_test)[0]->count;
       } else {
         $sql_test =
           'SELECT count(distinct r.rid) as count ' .
           'FROM run r, task t, algorithm_setup s ' .
           'WHERE r.setup = s.sid AND r.task_id = t.task_id ' .
-          $where_runs .
-          $where_limit ;
+          $where_runs;
```

Comment:
$where_limit is no longer used? Why not? if not necessary, remove entirely?

Reply:
Good point. The logic of these statements seems a bit flawed

Reply:
Nvm, I double checked an the logic seems fine.